ors-api/src/main/java/org/heigit/ors/api/EngineProperties.java

```diff
@@ -119,59 +185,170 @@ public String getCachePath() {
+            return profile;
+        }
+
+        public void setProfile(String profile) {
+            this.profile = profile;
+        }
+
+        public boolean isEnabled() {
+            return enabled != null && enabled;
+        }
+
+        public void setEnabled(boolean enabled) {
+            this.enabled = enabled;
+        }
 
         public boolean isElevation() {
-            return elevation;
+            return elevation != null && elevation;
         }
 
         public void setElevation(boolean elevation) {
             this.elevation = elevation;
         }
 
+        public Boolean getElevationSmoothing() {
+            return elevationSmoothing != null && elevationSmoothing;
+        }
+
+        public void setElevationSmoothing(Boolean elevationSmoothing) {
+            this.elevationSmoothing = elevationSmoothing;
+        }
+
         public boolean isTraffic() {
-            return traffic;
+            return traffic != null && traffic;
         }
 
         public void setTraffic(boolean traffic) {
             this.traffic = traffic;
         }
 
+        public boolean getInterpolateBridgesAndTunnels() {
+            return interpolateBridgesAndTunnels != null && interpolateBridgesAndTunnels;
+        }
+
+        public void setInterpolateBridgesAndTunnels(Boolean interpolateBridgesAndTunnels) {
+            this.interpolateBridgesAndTunnels = interpolateBridgesAndTunnels;
+        }
+
+        public boolean getInstructions() {
+            return instructions != null ? instructions : true;
+        }
+
+        public void setInstructions(Boolean instructions) {
+            this.instructions = instructions;
+        }
+
+        public boolean getOptimize() {
+            return optimize != null && optimize;
+        }
+
+        public void setOptimize(Boolean optimize) {
+            this.optimize = optimize;
+        }
+
+        public String getGraphPath() {
+            return graphPath;
+        }
+
+        public void setGraphPath(String graphPath) {
+            this.graphPath = graphPath;
+        }
+
         public Map<String, String> getEncoderOptions() {
             return encoderOptions;
         }
 
+        public String getEncoderOptionsString() {
+            if (encoderOptions == null || encoderOptions.size() == 0)
+                return "";
+            StringBuilder output = new StringBuilder();
+            for (Map.Entry<String, String> entry : encoderOptions.entrySet()) {
+                if (output.length() > 0) {
+                    output.append("|");
+                }
+                output.append(entry.getKey()).append("=").append(entry.getValue());
+            }
+            return output.toString();
+        }
+
         public void setEncoderOptions(Map<String, String> encoderOptions) {
             this.encoderOptions = encoderOptions;
         }
 
-        public PreparationProperties getPreparation() {
+//        public PreparationProperties getPreparation() {
```

Comment:
remove commented out code?

Reply:
I'm hoping that we are going to refactor the `RoutingProfileConfiguration` after we have removed the old AppConfig. The commented out code is actually better but will make sense only after we have also refactored how these parameters are passed and used.